libexec/pyenv-init

```diff
@@ -143,9 +150,17 @@ function init_dirs() {
 function print_path() {
   case "$shell" in
     fish )
-      echo 'while set index (contains -i -- '\'"${PYENV_ROOT}/shims"\'' $PATH)'
-      echo 'set -eg PATH[$index]; end; set -e index'
-      echo 'set -gx PATH '\'"${PYENV_ROOT}/shims"\'' $PATH'
+      echo 'while set pyenv_index (contains -i -- '\'"${PYENV_ROOT}/shims"\'' $PATH)'
+      if [ -n "$no_push_path" ]; then
+        echo 'set found_pyenv_shim 1; break; end';
+        echo 'if test -z "$found_pyenv_shim"'
+        echo 'set -gx PATH '\'"${PYENV_ROOT}/shims"\'' $PATH'
+        echo 'else; set -e found_pyenv_shim; end'
+      else
+        echo 'set -eg PATH[$pyenv_index]; end;'
+        echo 'set -gx PATH '\'"${PYENV_ROOT}/shims"\'' $PATH'
+      fi
+      echo 'set -e pyenv_index'
```

Comment:
This looks overcomplicated. It's probably better to split this into two chunks for either case.

Reply:
Yes, it's complicated.  I'm not sure what splitting into chunks accomplishes, however -- don't both branches need the same while loop?

Reply:
The `--no-push` version doesn't need a loop at all, only a one-time condition check.